### PR TITLE
Add Spring Security and layers for MVC

### PR DESCRIPTION
**Pull request by Vladlis**

### CODE REVIEW

File: web-project/src/main/java/com/epam/webproject/service/impl/BookServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import com.epam.webproject.service.BookService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service("bookService")
```
Comment (sqshq) [inline web-project/src/main/java/com/epam/webproject/service/impl/BookServiceImpl.java]: Why do we need hardcoded qualifier (here and other places)?

File: web-project/src/main/java/com/epam/webproject/dao/impl/BookDaoMock.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+@Repository
+public class BookDaoMock implements BookDao {
+
+    private Map<Integer, Book> storage = new HashMap<>();
+    private int id = 0;
```
Comment (sqshq) [inline web-project/src/main/java/com/epam/webproject/dao/impl/BookDaoMock.java]: It's ok for mock dao, but don't forget that it's not safe to use `HashMap` and `int` for modification in multithreaded environment

File: web-project/src/main/java/com/epam/webproject/controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/user")
+public class UserController {
+
+    @Autowired
+    private AuthorService authorService;
+
+    @RequestMapping("/getAuthor")
```
Comment (sqshq) [inline web-project/src/main/java/com/epam/webproject/controller/UserController.java]: I suggest to use REST naming conventions for url path (resources) and http methods to describe the action.

REST naming patterns/antipatterns: http://www.restapitutorial.com/lessons/restfulresourcenaming.html
Must-read about http: https://habrahabr.ru/company/yandex/blog/265569/

File: web-project/src/main/java/com/epam/webproject/config/SecurityConfig.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
+
+    @Autowired
+    public void configureGlobalSecurity(AuthenticationManagerBuilder auth) throws Exception {
+        auth.inMemoryAuthentication()
+                .withUser("user").password("user").roles("USER");
+        auth.inMemoryAuthentication()
```
Comment (sqshq) [inline web-project/src/main/java/com/epam/webproject/config/SecurityConfig.java]: you can use `.and()` here :)

File: web-project/src/main/java/com/epam/webproject/config/SecurityConfig.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+                .withUser("admin").password("admin").roles("ADMIN");
+    }
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http.authorizeRequests().antMatchers("/login").permitAll()
+                .antMatchers("/user/**").access("hasRole('USER')")
+                .antMatchers("/admin/**").access("hasRole('ADMIN')")
```
Comment (sqshq) [inline web-project/src/main/java/com/epam/webproject/config/SecurityConfig.java]: there is a handy method `.hasRole()` (which is a shortcut for `access("hasRole('...')")` actually

**Review by sqshq — CHANGES_REQUESTED** [review]

File: web-project/src/main/java/com/epam/webproject/service/impl/BookServiceImpl.java (hunk 1 above)
Comment (Vladlis) [reply web-project/src/main/java/com/epam/webproject/service/impl/BookServiceImpl.java]: Removed

File: web-project/src/main/java/com/epam/webproject/dao/impl/BookDaoMock.java (hunk 2 above)
Comment (Vladlis) [reply web-project/src/main/java/com/epam/webproject/dao/impl/BookDaoMock.java]: Fixed

File: web-project/src/main/java/com/epam/webproject/controller/UserController.java (hunk 3 above)
Comment (Vladlis) [reply web-project/src/main/java/com/epam/webproject/controller/UserController.java]: Done

File: web-project/src/main/java/com/epam/webproject/config/SecurityConfig.java (hunk 4 above)
Comment (Vladlis) [reply web-project/src/main/java/com/epam/webproject/config/SecurityConfig.java]: Fixed

File: web-project/src/main/java/com/epam/webproject/config/SecurityConfig.java (hunk 5 above)
Comment (Vladlis) [reply web-project/src/main/java/com/epam/webproject/config/SecurityConfig.java]: Done

**Review by sqshq — APPROVED** [review]

**Pull request merged**